shimmingtoolbox/load_nifti.py

```diff
@@ -216,37 +216,58 @@ def scale_tfl_b1(image, json_data):
         warnings.warn("Missing json tag: 'SliceTiming', slices number cannot be checked.")
         n_slices = image.shape[2]
 
-    # Calculate B1 efficiency (1ms, pi-pulse) and scale by the ratio of the measured FA to the saturation FA.
-    # Get the Transmission amplifier reference amplitude
-    amplifier_voltage = json_data['TxRefAmp']  # [V]
-    socket_voltage = amplifier_voltage * 10 ** -0.095  # -1.9dB voltage loss from amplifier to coil socket
     # Magnitude values are stored in the first half of the 4th dimension
-    b1_mag = image[:, :, :, :image.shape[3] // 2] / 10  # Siemens magnitude values are stored in degrees x10
+    b1_mag = image[:, :, :, :image.shape[3] // 2]
 
     if b1_mag.min() < 0:
         raise ValueError("Unexpected negative magnitude values")
 
+    # Phase values are stored in the second half of the 4th dimension. Siemens phase range: [248 - 3848]
+    b1_phase = image[:, :, :, image.shape[3] // 2:]
+
+    # Reorder data shuffled by dm2niix into shape (x, y , n_slices*n_coils)
+    b1_mag_vector = np.zeros((image.shape[0], image.shape[1], n_slices * n_coils))
+    b1_phase_vector = np.zeros((image.shape[0], image.shape[1], n_slices * n_coils))
+    if 'ImageOrientationPatientDICOM' in json_data:
+        # If axial slices
+        if json_data['ImageOrientationPatientDICOM'] == [1, 0, 0, 0, 1, 0]:
```

Comment:
Is this always going to be the case that axial = [1, 0, 0, 0, 1, 0]? What does the vector mean?

Reply:
It should be, as it correspond to this DICOM tag: https://dicom.innolitics.com/ciods/mr-image/image-plane/00200037

Reply:
So it's a direction, would that mean it could be -1?

Reply:
Yes I think for coronal images one value is -1. But from what I found on forums/doc there seem to be 3 cases that cover 99% of the scenarios. If I ever have an unexpected value I will just raise an error.